.github/workflows/build-and-test.yml

```
@@ -85,3 +85,21 @@ jobs:
           name: Test Reports
           path: service/build/reports
           retention-days: 14
+          
+  bump-check:
+    runs-on: ubuntu-latest
+    outputs:
+      is-bump: ${{ steps.skiptest.outputs.is-bump }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Skip version bump merges
+        id: bumpcheck
+        uses: ./.github/actions/bump-skip
+        with:
+          event-name: ${{ github.event_name }}
+
+  tag:
+    if: always() && needs.bump-check.outputs.is-bump == 'no' && github.event_name == 'push' && github.ref == 'refs/heads/main'
```

`always()` looks unnecessary here since there are other conditions.

```suggestion
    if: needs.bump-check.outputs.is-bump == 'no' && github.event_name == 'push' && github.ref == 'refs/heads/main'
```

---

I wondered about that!  I looked it up and the `always()` makes sure it runs (or checks to see if it should run) if previous steps fail.  I'm not sure if we want it to run if other steps fail or not, but it at least isn't as senseless as it seems at first.

---

Ah.
> You can use the following status check functions as expressions in if conditionals. A default status check of success() is applied unless you include one of these functions
https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions